components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/ApiPurge.java

```diff
@@ -131,8 +136,18 @@ private void setupPersistenceManager(){
                         break;
                     }
                     apiPurgeTaskMap.put(task.getKey(), APIConstants.OrganizationDeletion.COMPLETED);
+                    if (!isAPIOrganizationExist) {
+                        throw new APIManagementException("Organization: "+organization+" doesn't exist to perform the"
```

Comment:
This should not be return . APIManagementException..Add the message to apiPurgeTaskMap